mkosi.build

```diff
@@ -14,6 +14,18 @@ if ! mountpoint -q "$SRCDIR"; then
         umask 022
 fi
 
+# On Fedora "ld" is (unfortunately — if you ask me) managed via
+# "alternatives". Since we'd like to support building images in environments
+# with only /usr/ around (e.g. mkosi's UsrOnly=1 option), we have the problem
+# that /usr/bin/ld is a symlink that points to a non-existing file in
+# /etc/alternative/ in this mode. Let's work around this for now by manually
+# redirect "ld" to "ld.bfd", i.e. circumventing the /usr/bin/ld symlink.
+if [ ! -x /usr/bin/ld -a -x /usr/bin/ld.bfd ] ; then
+        mkdir -p "$HOME"/bin
```

Comment:
Would exporting an alias work as an alternative? Leaving stuff around in $HOME feels a bit off

Reply:
mkosi builds images twice: the first one contains all development deps, and is what this build script is run in. this script then builds systemd and generates output artifacts from that. We then build another image afresh, without the build deps, only with runtime deps, and with none of the build sources around or the build script or what they generated from the first run. The build artifacts from the first run are then copied into this second image, and that is then the final result that is supposed to be deployed.

Long story short: the build script can change $HOME as much as it wants, it won't end up in the final image. $HOME is never copied there, and on the next build we start fresh too. Any changes made to $HOME hence have a lifetime only until the build completes.

"alias" doesn't work, as that is internal to the shell. We need "ld" to work when called from meson/ninja though, i.e. a bunch of processes down thre tree...

Reply:
Ok, got it - still feels a bit hacky, but if there's no better way, let's go for it